animation_nodes/nodes/spline/transform_spline.py

```diff
@@ -1,22 +1,63 @@
 import bpy
+from mathutils import Matrix
+from ... data_structures import VirtualPyList, Spline
 from ... base_types import AnimationNode, VectorizedSocket
 
 class TransformSplineNode(bpy.types.Node, AnimationNode):
     bl_idname = "an_TransformSplineNode"
     bl_label = "Transform Spline"
-    codeEffects = [VectorizedSocket.CodeEffect]
 
     useSplineList: VectorizedSocket.newProperty()
+    useMatrixList: VectorizedSocket.newProperty()
 
     def create(self):
         socket = self.newInput(VectorizedSocket("Spline", "useSplineList",
             ("Spline", "spline"), ("Splines", "splines")))
         socket.dataIsModified = True
```

Comment:
`dataIsModified` is only needed when `useMatrixList == False`.

Reply:
Done